tuxemon/core/states/world/worldstate.py

```diff
@@ -388,56 +337,309 @@ def map_drawing(self, surface):
         :returns: None
 
         """
-        # center the camera on the player sprite
-        sx, sy = prepare.SCREEN_SIZE
-        self.current_map.renderer.center((-self.global_x + sx / 2,
-                                          -self.global_y + sy / 2))
-
+        # TODO: move all drawing into a "WorldView" widget
         # interlace player sprites with tiles surfaces.
         # eventually, maybe use pygame sprites or something similar
-        surfaces = self.player1.get_sprites()
+        world_surfaces = list()
+
+        # get player coords to center map
+        cx, cy = nearest(self.project(self.player1.tile_pos))
+
+        # offset center point for player sprite
```

Comment:
The player sprite is actually centered correctly now, but it actually introduced a rendering issue with grass tiles. It now clips the top of the player sprite.

Here's the current `development` branch:

**Old**
![](https://i.imgur.com/ZFdUTJl.png)

And what's in the PR:

**New**
![](https://i.imgur.com/c1QEEwj.png)